test/system/components/student/tracks_list_test.rb

```diff
@@ -3,54 +3,56 @@
 module Components
   module Student
     class TracksListTest < ApplicationSystemTestCase
-      include TableMatchers
-
       def setup
         super
 
-        # This component uses the API, which requires authentication
+        # This component uses the API, which requires authentication.
         sign_in!
       end
 
       test "shows tracks" do
-        track = create :track, title: "Ruby"
-        concept_exercises = Array.new(10).map { create :concept_exercise, track: track }
-        practice_exercises = Array.new(10).map { create :practice_exercise, track: track }
+        track = create :track,
+          title: "Ruby",
+          tags: ["Foo:Bar", "Abc:Xyz"]
+        concept_exercises = Array.new(3).map { create :concept_exercise, track: track }
+        practice_exercises = Array.new(4).map { create :practice_exercise, track: track }
         create :user_track, track: track, user: @current_user
-        concept_exercises.sample(5).map { |ex| create :concept_solution, user: @current_user, exercise: ex }
-        practice_exercises.sample(5).map { |ex| create :practice_solution, user: @current_user, exercise: ex }
+        concept_exercises.sample(1).map { |ex| create :concept_solution, user: @current_user, exercise: ex }
+        practice_exercises.sample(2).map { |ex| create :practice_solution, user: @current_user, exercise: ex }
 
         visit test_components_student_tracks_list_url
 
-        row = {
-          "Icon" => lambda {
-            assert_css "img[src='https://assets.exercism.io/tracks/ruby-hex-white.png'][alt='icon for Ruby track']"
-          },
-          "Title" => "Ruby",
-          "Num concept exercises" => 10,
-          "Num practice exercises" => 10,
-          "URL" => "https://test.exercism.io/tracks/ruby",
-          "New?" => "true",
-          "Tags" => "OOP, Web Dev",
-          "Joined?" => "true",
-          "Num completed concept exercises" => 5,
-          "Num completed practice exercises" => 5,
-          "Progress" => "Completed concept exercises: 25%, Uncompleted concept exercises: 25%, "\
-                        "Completed practice exercises: 25%, Uncompleted practice exercises: 25%"
-        }
-        assert_table_row first("table"), row
+        assert_html '
+          <a class="c-track" href="https://test.exercism.io/tracks/ruby">
+            <img class="c-track-icon"
+                 src="https://assets.exercism.io/tracks/ruby-hex-white.png"
+                 alt="icon for Ruby track">
+            <div class="info">
+              <h3 class="title">Ruby</h3>
+              <ul class="counts">
+                <li>1/3 concepts</li>
+                <li>2/4 exercises</li>
+              </ul>
+              <ul class="tags">
+                <li>Bar</li>
+                <li>Xyz</li>
+              </ul>
+            </div>
+            <i>›</i>
+          </a>
+        ', within: ".student-track-list"
```

Comment:
I started off by asserting each bit and descended into madness. I rethought it based on `assert_table_row` pattern, and decided having one assertion for everything made much more sense. 

So having one "happy-path" assertion of the rendered HTML, then using other methods to check bits that change. We'll probably want a second full-check for the joined track too.